contracts/Core/StakeManager.sol

```diff
@@ -181,23 +191,20 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         require(lock.withdrawAfter <= epoch, "Withdraw epoch not reached");
         require(lock.withdrawAfter + parameters.withdrawReleasePeriod() >= epoch, "Release Period Passed"); // Can Use ResetLock
 
-        uint256 lockedAmount = lock.amount;
-
-        // Function to Reset the lock
-        _resetLock(stakerId);
+        uint256 withdrawAmount = lock.amount;
 
-        // Transfer commission in case of delegators
-        // Check commission rate >0
-        if (stakerIds[msg.sender] != stakerId && staker.commission > 0) {
-            uint256 commission = (lockedAmount * staker.commission) / 100;
-            lockedAmount = lockedAmount - commission;
-            require(razor.transfer(staker._address, commission), "couldnt transfer");
+        if (lock.commission > 0) {
+            withdrawAmount = lock.amount - lock.commission;
+            require(razor.transfer(staker._address, lock.commission), "couldnt transfer");
         }
 
-        //Transfer stake
-        require(razor.transfer(msg.sender, lockedAmount), "couldnt transfer");
+        // Reset lock
+        _resetLock(stakerId);
```

Comment:
state change after transfer. this should be moved up

Reply:
resolved